pkg/miniogw/gateway-storj.go

```diff
@@ -95,12 +95,12 @@ func (s *storjObjects) GetObject(ctx context.Context, bucket, object string,
 	startOffset int64, length int64, writer io.Writer, etag string) (err error) {
 	defer mon.Task()(&ctx)(&err)
 	objpath := paths.New(bucket, object)
-	rr, _, err := s.storj.os.GetObject(ctx, objpath)
+	rr, _, err := s.storj.os.Get(ctx, objpath)
 	if err != nil {
 		return err
 	}
 	defer func() {
-		if err := rr.Close(); err != nil {
+		if err = rr.Close(); err != nil {
```

Comment:
oh man this is a huge bug. i almost want to figure out how to get rid of errcheck for these defer issues.

it turns out that the colon you removed was load bearing. if you have the colon, inside the defer you're defining a new err variable and checking it. without the colon, you're now overwriting the return value, which is named err!!

summary: with a colon, this is fine. without a colon, this is a huge bug that's going to cause very hard to debug silent error swallowing.

@coyle this is a huge footgun. what do you think? should we try and get errcheck to ignore deferred calls?

Reply:
@aleitner defined `utils.Close()` helper in PR #134: https://github.com/storj/storj/pull/134/files#diff-a0fc870d3e98cb9e7ffa9fcb9fb1d041R39

Once it's merge we can just call `defer utils.Close(rr)` here.